### ISSUE TITLE
Table headers do not paint borders and separators when it is a custom wrapper

### ISSUE THREAD

**Issue by Chrriis** [JFormDesigner/FlatLaf#37]
Hi,

It is common to tweak default behaviors by creating wrappers that delegate to the original renderer.
When we do this with table headers in FlatLaf, then borders and separators are not painted anymore.

Here is sample code showing two tables, one standard and one with delegate:
```Java
    public static void main(String[] args) throws Exception {
        UIManager.setLookAndFeel(new FlatLightLaf());
        JFrame frame = new JFrame();
        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
        JPanel centerPane = new JPanel(new GridLayout(2, 1));
        JTable table1 = new JTable(new Object[][] {{"Data_1_1", "Data_1_2"}, {"Data_2_1", "Data_2_2"}}, new String[] {"Col1", "Col2"});
        centerPane.add(new JScrollPane(table1));
        JTable table2 = new JTable(new Object[][] {{"Data_1_1", "Data_1_2"}, {"Data_2_1", "Data_2_2"}}, new String[] {"Col1", "Col2"});
        JTableHeader tableHeader = table2.getTableHeader();
        TableCellRenderer defaultRenderer = tableHeader.getDefaultRenderer();
        tableHeader.setDefaultRenderer(new TableCellRenderer() {
            @Override
            public Component getTableCellRendererComponent(JTable table, Object value, boolean isSelected, boolean hasFocus, int row, int column) {
                return defaultRenderer.getTableCellRendererComponent(table, value, isSelected, hasFocus, row, column);
            }
        });
        centerPane.add(new JScrollPane(table2));
        frame.getContentPane().add(centerPane);
        frame.setSize(400, 300);
        frame.setVisible(true);
    }
```

And here is a screenshot of the result:
![TableHeaderOverride](https://user-images.githubusercontent.com/1174202/72082444-75f18300-3300-11ea-92e4-6fbd4864505e.png)

The problem is because of magic code in "FlatTableHeaderUI.paint(g, c)" based on renderer class names.

**Comment by DevCharly** [comment]
Yes, have noticed this also some days ago. It is already [fixed](https://github.com/JFormDesigner/FlatLaf/commit/8dbbe208404f85ad17873b25312eb5b24ba80b32) in master branch. Next release is probably coming tomorrow.

There is now even more magic code 😉

**Comment by Chrriis** [comment]
I can confirm that it is fixed in master!

Indeed, more magic in the code. I wonder if it can break or cause problems, but it seems to work as expected.

Thanks a lot!

**Comment by DevCharly** [comment]
fixed in 0.24
